clients/fides-js/src/lib/consent-types.ts

```diff
@@ -207,6 +207,49 @@ export type UserGeolocation = {
 // 3) Separated by a dash (e.g. "US-CA")
 export const VALID_ISO_3166_LOCATION_REGEX = /^\w{2,3}(-\w{2,3})?$/;
 
+export type OverrideOptions = {
+  fides_string: string;
+  fides_disable_save_api: boolean;
+  fides_embed: boolean;
+};
+
+export type FidesOptionOverrides = Pick<
+  FidesOptions,
+  "fidesString" | "fidesDisableSaveApi" | "fidesEmbed"
+>;
+
+export enum OverrideFidesOption {
```

Comment:
yes I know I've added a lot of types / enums 😅 , but I'm open to a better way to consolidate. Was fighting with the linter a bit here.

Reply:
I don't think you need the enum—since you type to a list of strings by doing a `keyof`, we get just as much type safety as we would with an enum. so that can save you one type: 

```typescript
export type OverrideOptions = {
  fides_string: string;
  fides_disable_save_api: boolean;
  fides_embed: boolean;
};

export type FidesOptionOverrides = Pick<
  FidesOptions,
  "fidesString" | "fidesDisableSaveApi" | "fidesEmbed"
>;

export const FIDES_OVERRIDE_OPTIONS_VALIDATOR_MAP: {
  fidesOption: keyof FidesOptionOverrides;
  fidesOptionType: "string" | "boolean";
  fidesOverrideKey: keyof OverrideOptions;
  validationRegex: RegExp;
}[] = [
  {
    fidesOption: "fidesEmbed",
    fidesOptionType: "boolean",
    fidesOverrideKey: "fides_embed",
    validationRegex: /^(true|false)$/,
  },
  {
    fidesOption: "fidesDisableSaveApi",
    fidesOptionType: "boolean",
    fidesOverrideKey: "fides_disable_save_api",
    validationRegex: /^(true|false)$/,
  },
  {
    fidesOption: "fidesString",
    fidesOptionType: "string",
    fidesOverrideKey: "fides_string",
    validationRegex: /(.*)/,
  },
];
```

and if you're okay with camelcasing everything, you can bring that down one type farther to just the one type, like so...

```typescript

export type FidesOptionOverrides = Pick<
  FidesOptions,
  "fidesString" | "fidesDisableSaveApi" | "fidesEmbed"
>;

export const FIDES_OVERRIDE_OPTIONS_VALIDATOR_MAP: {
  fidesOption: keyof FidesOptionOverrides;
  fidesOptionType: "string" | "boolean";
  validationRegex: RegExp;
}[] = [
  {
    fidesOption: "fidesEmbed",
    fidesOptionType: "boolean",
    validationRegex: /^(true|false)$/,
  },
  {
    fidesOption: "fidesDisableSaveApi",
    fidesOptionType: "boolean",
    validationRegex: /^(true|false)$/,
  },
  {
    fidesOption: "fidesString",
    fidesOptionType: "string",
    validationRegex: /(.*)/,
  },
];
```